File: cnt_diff.py
import statistics

def cnt_diff(data_in, data_points):
    data_in_diff = []
    for i in range(1, data_points):
        data_in_diff.append(data_in[i] - data_in[i - 1])
    data_in_med = statistics.median(data_in_diff)
    ind = []
    indLen = 0
    if data_in_med > 2:
        for i in range(0, data_points - 1):
            if data_in_diff[i] < 1.5 * data_in_med and data_in_diff[i] > 0.5 * data_in_med:
                ind.append(i)
                indLen = indLen + 1
    if data_in_med <= 1:
        for i in range(0, data_points - 1):
            if data_in_diff[i] <= 2 * data_in_med and data_in_diff[i] >= 0:
                ind.append(i)
                indLen = indLen + 1
    if data_in_med <= 2 and data_in_med > 1:
        for i in range(0, data_points - 1):
            if data_in_diff[i] <= 2 * data_in_med and data_in_diff[i] >= 0:
                ind.append(i)
                indLen = indLen + 1
    sum = 0
    for i in range(0, indLen):
        sum = data_in_diff[ind[i]] + sum
    cnt_avg = sum / indLen
    return cnt_avg

File: cnt_diff.m
function cnt_avg = cnt_diff(data_in)
    data_in_diff = diff(data_in);
    data_in_med = median(data_in_diff);
    if (data_in_med >2)
    ind = find(data_in_diff<1.5*data_in_med & data_in_diff>0.5*data_in_med ); 
    end
    if (data_in_med <=1)
        ind = find(data_in_diff<=2*data_in_med & data_in_diff>=0 ); 
    end
    if (data_in_med <=2 && data_in_med > 1)
        ind = find(data_in_diff<=2*data_in_med & data_in_diff>=0 ); 
    end
    cnt_avg = mean(data_in_diff(ind)); 
    data_final = data_in_diff(ind);
    std(data_final);
end